examples/hosts/app-integration/external-data/src/utilities/clientManager.ts

```diff
@@ -0,0 +1,106 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// import fetch from "node-fetch";
+
+/**
+ * Represents a Fluid client URL.
+ * This URL contains the client's Fluid session information necessary for broadcasting signals to.
+ */
+export type ClientSessionUrl = string;
+
+/**
+ * Represents the external data servers query url or uuid.
+ * This is the URL or the id of the external resource that the customer service needs to subscribe for at the external service.
+ */
+export type ExternalTaskListId = string;
+
+/**
+ * Mock client manager.
+ * Can be registered with to receive notifications when {@link ClientManager.notifySubscribers} is called.
+ *
+ * @typeParam TData - The kind of data that will be sent alongside notifications to subscribers.
+ * Must be JSON-serializable.
+ */
+export class ClientManager<TData = unknown> {
+	/**
+	 * Map of active external query id to client session.
+	 * Values are the URLs that will be notified of changes.
+	 */
+	private readonly _taskListMapping: Map<ExternalTaskListId, Set<ClientSessionUrl>>;
+	/**
+	 * Map of active clients to external query id.
+	 * Values are the URLs that will be notified of changes.
+	 */
+	private readonly _clientMapping: Map<ClientSessionUrl, Set<ExternalTaskListId>>;
```

Comment:
Not sure exactly what its meant by "external query id" here. I was thinking purely an ID, but some of these docs suggest they are full URLs? Which kind of matches the docs for the `ExternalTaskListId` (which are a bit ambiguous, however: "Represents the external data servers query url or uuid."; is it a URL or a UUID?), but usage of the type in `externalDataSource.ts` makes me think its just an id.

Reply:
Changing the word `query` to resource. Fo this example, I've definitely been treating them as id's.